sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.applicationCancelChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,200 @@
+              modifier: { id: true },
+              updatedAt: true,
+            },
+          },
+          relations: {
+            applicationEditStatusUpdatedBy: true,
+            modifier: true,
+            currentAssessment: {
+              offering: true,
+              studentAppeal: true,
+              modifier: true,
+            },
+          },
+          where: {
+            id: completedApplication.id,
+          },
+          loadEagerRelations: false,
+        });
+        // Expected user object.
+        const expectedUserObject = expect.objectContaining({
+          id: student.user.id,
+        });
+        // Expected updated fields.
+        expect(cancelledChangeRequestApplication).toEqual({
+          id: completedApplication.id,
+          applicationEditStatus: ApplicationEditStatus.ChangeCancelled,
+          applicationEditStatusUpdatedOn: now,
+          applicationEditStatusUpdatedBy: expectedUserObject,
+          modifier: expectedUserObject,
+          updatedAt: now,
+          currentAssessment: expect.objectContaining({
+            id: completedApplication.currentAssessment.id,
+            studentAssessmentStatus:
+              StudentAssessmentStatus.CancellationRequested,
+            studentAssessmentStatusUpdatedOn: now,
+            modifier: expectedUserObject,
+            updatedAt: now,
+          }),
+        });
+      });
+    });
+  });
+
+  it("Should throw a NotFoundException when the application has already been declined by the Ministry.", async () => {
+    // Arrange
+    const completedApplication = await saveFakeApplication(
+      db.dataSource,
+      undefined,
+      {
+        applicationStatus: ApplicationStatus.Edited,
+        applicationEditStatus: ApplicationEditStatus.ChangeDeclined,
+      },
+    );
+    const endpoint = `/students/application/${completedApplication.id}/cancel-change-request`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    await mockUserLoginInfo(appModule, completedApplication.student);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        message: "Not able to find the in-progress change request.",
+        error: "Not Found",
+        statusCode: HttpStatus.NOT_FOUND,
+      });
+  });
+
+  it("Should throw a NotFoundException when the application does not belong to the student.", async () => {
+    // Arrange
+    const completedApplication = await saveFakeApplication(
+      db.dataSource,
+      undefined,
+      {
+        applicationStatus: ApplicationStatus.Edited,
+        applicationEditStatus: ApplicationEditStatus.ChangeInProgress,
+      },
+    );
+    const endpoint = `/students/application/${completedApplication.id}/cancel-change-request`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
```

Comment:
A minor observation here. This test currently works because the mock is set by previous tests to use different studentId and I believe the mock is not cleared not because of `FakeStudentUserType1` user is returned at runtime. Not a blocker.

Reply:
It works because the student associated with the application will never be the one mocked before, or the one if we reset the current mock. I can reset the mock if it would make it clear.

Reply:
Reset wouldn't be required in this case, If there was a shared student then it would be required.